pkg/apis/devops/v1alpha2/register.go

```diff
@@ -0,0 +1,119 @@
+	webservice.Route(webservice.GET("/devops/{projectName}/pipelines/{pipelineName}/runs").
+		To(devops.SearchPipelineRuns).
+		Metadata(restfulspec.KeyOpenAPITags, tags).
+		Doc("Search DevOps Pipelines runs.").
+		Param(webservice.PathParameter("pipelineName", "pipeline name")).
+		Param(webservice.PathParameter("projectName", "devops project name")).
+		Param(webservice.QueryParameter("start", "start page").
+			Required(false)).
+		Param(webservice.QueryParameter("limit", "limit count").
+			Required(false)))
+
+	// match Jenkins api "/blue/rest/organizations/jenkins/pipelines/{projectName}/{pipelineName}/branches/{branchName}/runs/{runId}/"
+	webservice.Route(webservice.GET("/devops/{projectName}/pipelines/{pipelineName}/branches/{branchName}/runs/{runId}").
+		To(devops.GetPipelineRun).
+		Metadata(restfulspec.KeyOpenAPITags, tags).
+		Doc("Get DevOps Pipelines run.").
+		Param(webservice.PathParameter("pipelineName", "pipeline name")).
+		Param(webservice.PathParameter("projectName", "devops project name")).
+		Param(webservice.PathParameter("branchName", "pipeline branch name")).
+		Param(webservice.PathParameter("runId", "pipeline runs id")))
+
+	// match Jenkins api "/blue/rest/organizations/jenkins/pipelines/{projectName}/{pipelineName}/branches/{branchName}/runs/{runId}/nodes"
+	webservice.Route(webservice.GET("/devops/{projectName}/pipelines/{pipelineName}/branches/{branchName}/runs/{runId}/nodes").
+		To(devops.GetPipelineRunNodes).
+		Metadata(restfulspec.KeyOpenAPITags, tags).
+		Doc("Get node on DevOps Pipelines run.").
+		Param(webservice.PathParameter("projectName", "devops project name")).
+		Param(webservice.PathParameter("pipelineName", "pipeline name")).
+		Param(webservice.PathParameter("branchName", "pipeline branch name")).
+		Param(webservice.PathParameter("runId", "pipeline runs id")).
+		Param(webservice.QueryParameter("limit", "limit").
+			Required(false).
+			DataFormat("limit=%d").
+			DefaultValue("limit=10000")))
+
+	// match "/blue/rest/organizations/jenkins/pipelines/{projectName}/{pipelineName}/branches/{branchName}/runs/{runId}/nodes/{nodeId}/steps/{stepId}/log/?start=0"
+	webservice.Route(webservice.GET("/devops/{projectName}/pipelines/{pipelineName}/branches/{branchName}/runs/{runId}/nodes/{nodeId}/steps/{stepId}/log").
+		To(devops.GetStepLog).
+		Metadata(restfulspec.KeyOpenAPITags, tags).
+		Doc("Get Pipelines step log.").
+		Param(webservice.PathParameter("projectName", "devops project name")).
+		Param(webservice.PathParameter("pipelineName", "pipeline name")).
+		Param(webservice.PathParameter("branchName", "pipeline branch name")).
+		Param(webservice.PathParameter("runId", "pipeline runs id")).
+		Param(webservice.PathParameter("nodeId", "pipeline runs node id")).
+		Param(webservice.PathParameter("stepId", "pipeline runs step id")).
+		Param(webservice.QueryParameter("start", "start").
+			Required(true).
+			DataFormat("start=%d").
+			DefaultValue("start=0")))
```

Comment:
should dataformat applied just for value? I think default value is `0`, not `start=0`

Reply:
like this: 
```
{
            "type": "string",
            "format": "start=%d",
            "default": "start=0",
            "description": "start",
            "name": "start",
            "in": "query",
            "required": true
}
```
I think `start=0` is more completely and clearly. The previous Document is also this format。

Reply:
The key is fixed. I think no need to format the key.